### DISCUSSION TITLE
brew update & brew update-reset don't play well with zdiff3

### DISCUSSION THREAD

**Discussion by ahmedelgabri** [Homebrew/discussions#3304] [Getting started] [answered]
I'm having issues with homebrew updating when by setting `conflictStyle` to `zdiff3` (_[released in git 2.35](https://github.blog/2022-01-24-highlights-from-git-2-35/)_), I usually get errors like this.

<img width="378" alt="Screenshot 2022-05-24 at 21 39 30" src="https://user-images.githubusercontent.com/63876/170118659-ccd7b961-cdd0-457b-984a-049b7f27174d.png">

When I try to force it to update `brew update-reset`, I get the same errors & basically I have to manually cd to each directory and update the git history.

<img width="531" alt="Screenshot 2022-05-24 at 21 39 48" src="https://user-images.githubusercontent.com/63876/170119020-202b523f-5e35-4673-b481-38e85fbd1da6.png">

**Comment by alebcay** [comment, marked as answer]
Under most circumstances, Homebrew will use your system's `git` (`/usr/bin/git`). Thus, if you have this new choice of `conflictStyle` set in `.gitconfig`, it's possible that the older `git` version doesn't work with it (which is probably what you're seeing).

If you want to, you can try forcing the use of the `git` that Homebrew ships (which is considerably newer) by setting the environment variable `HOMEBREW_FORCE_BREWED_GIT`, and it may work for you.

**Reply by ahmedelgabri** [reply]
Thanks, @alebcay `HOMEBREW_FORCE_BREWED_GIT` worked.
